### PR TITLE
[Minor] Add .after() callback

### PR DESCRIPTION
**Pull request by ealush**
| Q                | A
| ---------------- | ---
| Bug fix?         | no
| New feature?     | yes
| Breaking change? | no
| Deprecations?    | no
| Documentation?   | yes
| Tests added?     | yes

## Change details

The after callback is a function that can be chained to a passable suite and allows invoking a callback whenever a certain field has finished running, regardless of whether it passed or failed. It accepts two arguments: `fieldName` and `callback`. You may chain multiple callbacks to the same field. When running, the `.after()` function passes down to its callback argument the current result object, **note** it might not be final yet, as there may be other async fields still being processed.

```js
import passable, { enforce } from 'passable';
const email_regex = /[A-Z0-9._%+-]+@[A-Z0-9.-]+\.[A-Z]{2,4}/gi;

passable('SendEmailForm', (test) => {

  test('UserEmail', 'must be a valid email', () => {
    enforce(data.email).matches(email_email_regex);
  });

  test('UserName', 'must not be blacklisted', new Promise((resolve, reject) => {
    fetch(`/isBlacklisted?name=${data.name}`).then(...).then((data) => {
      if (data.blackListed) {
        reject();
      } else {
        resolve();
      }
    });
  }));

  test('Content', 'must be between 5 and 500 chars', () => {
    enforce(data.content).largerThan(4).smallerThan(501);
  });
}).after((res) => {
  if (res.hasErrors('username')) {
    showUserNameErrors(res.errors)
  }
});
```

### CODE REVIEW

File: src/core/Passable/async.spec.js [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+import passable from '../..';
+
+const p = () => passable(lorem.word(), (test) => {
+    test('f1', 'should someting', new Promise((resolve) => setTimeout(resolve, 250)));
+
+    test('f1', 'should someting1', new Promise((resolve) => setTimeout(resolve, 2500)));
+});
+
+describe('Test async-done behavior', () => {
+    it('Should something', () => {
```
Comment (sahariko) [inline src/core/Passable/async.spec.js]: What is this test about?

File: src/core/ResultObject/index.js [hunk 2]
```diff
@@ -155,8 +179,13 @@ class ResultObject {
...
     */
     markAsDone(fieldName: string) {
-        if (this.async && this.async[fieldName]) {
-            this.async[fieldName] = { done: true };
+        if (this.async !== null && this.async[fieldName]) {
+            this.async[fieldName].done = true;
+
+            // run field callbacks set in `after`
+            if (this.async[fieldName].callbacks) {
+                this.async[fieldName].callbacks.forEach((callback) => callback(this), this);
```
Comment (sahariko) [inline src/core/ResultObject/index.js]: Do you really need the `this` argument?

Comment (ealush) [reply src/core/ResultObject/index.js]: No. Added test to reflect that.

File: src/core/Passable/async.spec.js (hunk 1 above)
Comment (ealush) [reply src/core/Passable/async.spec.js]: About nothing. Deleted.

**Pull request merged**